src/shared/components/CloudUpgradeButton.tsx

```diff
@@ -15,19 +15,22 @@ import CloudOnly from 'src/shared/components/cloud/CloudOnly'
 
 // Utils
 import {shouldShowUpgradeButton} from 'src/me/selectors'
+import {isFlagEnabled} from 'src/shared/utils/featureFlag'
 
 interface OwnProps {
-  className?: string
   buttonText?: string
-  size?: ComponentSize
+  className?: string
+  hidePromoMessage?: boolean
```

Comment:
We are doing it this way to minimize the number of changes. Instead of asking all consumers of `CloudUpgradeButton` to pass a new prop, we are only asking those components who need to opt-out of the promo messaging (**even when the feature is on, to avoid redundancy**) to pass a new prop.

Reply:
Yeah, this makes sense.

Originally, I avoided `showPromoMessage` as the prop name because when it is undefined, it has be defaulted to true.

This seems a bit unexpected for a prop, because why should a prop be truthy when it isn't passed in. Especially for a prop that is considered optional. In other words, here is an optional prop that shows something. In my mind that means, if i don't use the prop, i don't want that something shown. But surprise! That's not case. This optional prop will default to true if you don't pass it in.

But i guess that's not really that big of an issue. We will go with `showPromoMessage`